app/manifest_lock_digests_cmd.go

```diff
@@ -0,0 +1,167 @@
+						arch := "amd64"
+						if len(plat.Attrs) > 1 {
+							arch = plat.Attrs[1]
+						}
+						ref = manifest.ParseReference(ref.Name + "-" + mc.Version[0])
+						pkg, err := res.ResolveWithConfig(l, manifest.ExactSelector(ref), manifest.DarwinConfig(res.GetConfig(), arch))
+						if err != nil {
+							return errors.WithStack(err)
+						}
+						checksum, err := getSHASum(l, state, pkg, &ref)
+						if err != nil {
+							return errors.WithStack(err)
+						}
+						localmanifest.Manifest.SHA256Sums[pkg.Source] = checksum
+						//plat.Layer.SHA256 = checksum
+
+					}
+					if plat.Attrs[0] == "linux" {
+						ref = manifest.ParseReference(ref.Name + "-" + mc.Version[0])
+						pkg, err := res.ResolveWithConfig(l, manifest.ExactSelector(ref), manifest.LinuxConfig(res.GetConfig()))
+						if err != nil {
+							return errors.WithStack(err)
+						}
+						checksum, err := getSHASum(l, state, pkg, &ref)
+						if err != nil {
+							return errors.WithStack(err)
+						}
+						//plat.Layer.SHA256 = checksum
+						localmanifest.Manifest.SHA256Sums[pkg.Source] = checksum
+
+					}
+				}
+			} else if len(mc.Layer.Darwin) > 0 {
+				// an example is :https://github.com/cashapp/hermit-packages/blob/31f421d7396046f5fd296daa9239ecd1e2ba1d4b/openjdk.hcl#L33-L40
+				ref = manifest.ParseReference(ref.Name + "-" + mc.Version[0])
+				pkg, err := res.ResolveWithConfig(l, manifest.ExactSelector(ref), manifest.DarwinConfig(res.GetConfig(), "darwin"))
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				checksum, err := getSHASum(l, state, pkg, &ref)
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				localmanifest.Manifest.SHA256Sums[pkg.Source] = checksum
+
+			} else if len(mc.Layer.Linux) > 0 {
+				// an example is :https://github.com/cashapp/hermit-packages/blob/31f421d7396046f5fd296daa9239ecd1e2ba1d4b/openjdk.hcl#L33-L40
+				ref = manifest.ParseReference(ref.Name + "-" + mc.Version[0])
+				pkg, err := res.ResolveWithConfig(l, manifest.ExactSelector(ref), manifest.DarwinConfig(res.GetConfig(), "linux"))
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				checksum, err := getSHASum(l, state, pkg, &ref)
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				localmanifest.Manifest.SHA256Sums[pkg.Source] = checksum
+			} else {
+				// only version definitions
+				ref = manifest.ParseReference(ref.Name + "-" + mc.Version[len(mc.Version)-1])
+				pkg, err := res.Resolve(l, manifest.ExactSelector(ref))
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				checksum, err := getSHASum(l, state, pkg, &ref)
+				if err != nil {
+					return errors.WithStack(err)
+				}
+				localmanifest.Manifest.SHA256Sums[pkg.Source] = checksum
+			}
+
+		}
+
+		value, _ := hcl.Marshal(localmanifest.Manifest)
+
+		//Now just write down fresh manifest files in a subdirectory in the `bin`.
+		os.MkdirAll(filepath.Join(env.BinDir(), "lockedDigests"), os.ModePerm)
```

Comment:
Rename `lockedDigests` to `hermit.lock`.

Also, check for errors here and elsewhere. You can use `errors.Is(err, fs.ErrExist)` to ignore "already exists" errors.

Reply:
done